src/map/zone.cpp

```diff
@@ -1032,10 +1032,9 @@ void CZone::CharZoneIn(CCharEntity* PChar)
         PChar->StatusEffectContainer->DelStatusEffectSilent(EFFECT_MOUNTED);
     }
 
-    if (PChar->m_Costume != 0)
+    if (PChar->StatusEffectContainer->HasStatusEffect(EFFECT_COSTUME))
     {
-        PChar->m_Costume = 0;
```

Comment:
If this is being stored on the char, shouldn't it still be reset here?

Reply:
I can leave it in but it seems to always be 0 by this point in time which is why the previous check was always failing and never removed the status effect.

Reply:
Losing the effect sets it to zero in `onEffectLose`